tests/test_basic_sweeper.py

```diff
@@ -54,3 +58,42 @@ def test_split(
     )
     lret = [list(x) for x in ret]
     assert lret == expected
+
+
+def test_partial_failure(
+    monkeypatch: Any,
+    tmpdir: Any,
+) -> None:
+    monkeypatch.chdir("tests/test_apps/app_can_fail")
+    cmd = [
+        sys.executable,
+        "-Werror",
+        "my_app.py",
+        "--multirun",
+        "+divisor=1,0,2",
+        "hydra.run.dir=" + str(tmpdir),
+    ]
+    with Popen(cmd, stdout=PIPE, stderr=PIPE) as p:
+        stdout, stderr = p.communicate()
+        err = stderr.decode("utf-8").rstrip().replace("\r\n", "\n")
+        out = stdout.decode("utf-8").rstrip().replace("\r\n", "\n")
+        assert p.returncode != 0
+
```

Comment:
Add a comment explaining why you are not using run_python_script(), run_process or run_with_error().